ballerina/provider.bal

```diff
@@ -0,0 +1,137 @@
+            validation: connectionConfig.validation
+        };
+        chat:Client|error llmClient = new (azureAiConfig, serviceUrl);
+        if llmClient is error {
+            return error ai:Error("Failed to initialize AzureOpenAiProvider", llmClient);
+        }
+        self.llmClient = llmClient;
+        self.deploymentId = deploymentId;
+        self.apiVersion = apiVersion;
+    }
+
+    # Sends a chat request to the OpenAI model with the given messages and tools.
+    #
+    # + messages - List of chat messages 
+    # + tools - Tool definitions to be used for the tool call
+    # + stop - Stop sequence to stop the completion
+    # + return - Function to be called, chat response or an error in-case of failures
+    isolated remote function chat(ai:ChatMessage[] messages, ai:ChatCompletionFunctions[] tools, string? stop = ())
+        returns ai:ChatAssistantMessage|ai:LlmError {
+        chat:CreateChatCompletionRequest request = {stop, messages: self.mapToChatCompletionRequestMessage(messages)};
+        if tools.length() > 0 {
+            request.functions = tools;
+        }
+        chat:CreateChatCompletionResponse|error response =
+            self.llmClient->/deployments/[self.deploymentId]/chat/completions.post(self.apiVersion, request);
+        if response is error {
+            return error ai:LlmConnectionError("Error while connecting to the model", response);
+        }
+
+        record {|
+            chat:ChatCompletionResponseMessage message?;
+            chat:ContentFilterChoiceResults content_filter_results?;
+            int index?;
+            string finish_reason?;
+            anydata...;
+        |}[]? choices = response.choices;
+
+        if choices is () || choices.length() == 0 {
+            return error ai:LlmInvalidResponseError("Empty response from the model when using function call API");
+        }
+        chat:ChatCompletionResponseMessage? message = choices[0].message;
+        ai:ChatAssistantMessage chatAssistantMessage = {role: ai:ASSISTANT, content: message?.content};
+        chat:ChatCompletionFunctionCall? function_call = message?.function_call;
+        if function_call is chat:ChatCompletionFunctionCall {
+            chatAssistantMessage.toolCalls = [{name: function_call.name, arguments: function_call.arguments}];
+        }
+        return chatAssistantMessage;
+    }
+
+    private isolated function mapToChatCompletionRequestMessage(ai:ChatMessage[] messages)
+        returns chat:ChatCompletionRequestMessage[] {
+        chat:ChatCompletionRequestMessage[] chatCompletionRequestMessages = [];
+        foreach ai:ChatMessage message in messages {
+            if message is ai:ChatAssistantMessage {
+                chat:ChatCompletionRequestMessage assistantMessage = {role: ai:ASSISTANT};
+                ai:FunctionCall[]? toolCalls = message.toolCalls;
+                if toolCalls is ai:FunctionCall[] {
+                    assistantMessage["function_call"] = toolCalls[0];
```

Comment:
I think we need to use the tool_call field in here, since function_call is depreciated.
Lets address this in a separate PR